packages/password-input/src/InputIcon/InputIcon.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+
+import { cx } from '@leafygreen-ui/emotion';
+import CheckmarkIcon from '@leafygreen-ui/icon/dist/Checkmark';
+import WarningIcon from '@leafygreen-ui/icon/dist/Warning';
+import ErrorIcon from '@leafygreen-ui/icon/dist/X';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+
+import { States } from '../PasswordInput/PasswordInput.types';
+
+import {
+  baseStyles,
+  disabledStyles,
+  sizeStyles,
+  themeStateStyles,
+} from './InputIcon.styles';
+import { InputIconProps, StateProps } from './InputIcon.types';
+
+const validationIcons: Record<StateProps, React.ComponentType<any>> = {
+  [States.Error]: ErrorIcon,
+  [States.Warning]: WarningIcon,
+  [States.Valid]: CheckmarkIcon,
+};
+
+export const InputIcon = ({ state, size, disabled }: InputIconProps) => {
+  const ValidationIcon = validationIcons[state];
+
+  const { theme } = useDarkMode();
+
+  return (
+    <ValidationIcon
+      className={cx(
```

Comment:
do we need `role="presentation"`?

Reply:
yes!

Reply:
or not! the icons already have `role='img'` which apparently is correct for SVGs https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/img_role#svg_and_roleimg